unit_tests/test_scan_operation_thread.py

```diff
@@ -151,80 +153,70 @@ def test_negative_prometheus_validation_error(events, cluster):
-    if execute_mock == 'execute_async':
-        connection = ExecuteAsyncOperationTimedOutMockCqlConnectionPatient()
-    else:
-        connection = ExecuteOperationTimedOutMockCqlConnectionPatient()
-    db_cluster = DBCluster(connection, [node], {})
-    node.parent_cluster = db_cluster
-    default_params = ThreadParams(
-        db_cluster=db_cluster,
-        ks_cf='a.b',
-        mode=mode,
-        full_scan_aggregates_operation_limit=timeout,
-        full_scan_operation_limit=timeout,
-        **DEFAULT_PARAMS
-    )
-    with events.wait_for_n_events(events.get_events_logger(), count=2, timeout=10):
-        ScanOperationThread(default_params)._run_next_operation()  # pylint: disable=protected-access
-    all_events = get_event_log_file(events)
-    assert "Severity.NORMAL" in all_events[0] and "period_type=begin" in all_events[0]
-    assert f"Severity.{severity}" in all_events[1] and "period_type=end" in all_events[1]
+class ReadTimeoutConnection(MockCqlConnectionPatient):
+    def execute(*args, **kwargs):
+        # pylint: disable=unused-argument
+        # pylint: disable=no-method-argument
+        raise ReadTimeout("Operation timed out")
 
 
-class ExecuteReadTimeoutMockCqlConnectionPatient1(MockCqlConnectionPatient):
-    def execute(*args, **kwargs):
+class AsyncReadTimeoutConnection(MockCqlConnectionPatient):
+    def execute_async(*args, **kwargs):
         # pylint: disable=unused-argument
         # pylint: disable=no-method-argument
         raise ReadTimeout("Operation timed out")
 
 
-class ExecuteReadTimeoutMockCqlConnectionPatient2(MockCqlConnectionPatient):
+class ConnectionException(Exception):
+    pass
+
+
+class ConnectionExceptionConnection(MockCqlConnectionPatient):
     def execute(*args, **kwargs):
         # pylint: disable=unused-argument
         # pylint: disable=no-method-argument
-        raise ReadTimeout("some another reason")
+        raise ConnectionException("'Host has been marked down or removed'")
+
+
+class AsyncConnectionExceptionConnection(MockCqlConnectionPatient):
+    def execute_async(*args, **kwargs):
+        # pylint: disable=unused-argument
+        # pylint: disable=no-method-argument
+        raise ConnectionException("'Host has been marked down or removed'")
 
 
-@pytest.mark.parametrize(('execute_mock', "expected_message"),
-                         [[ExecuteReadTimeoutMockCqlConnectionPatient1, "operation failed due to operation timed out"],
-                          [ExecuteReadTimeoutMockCqlConnectionPatient2, "operation failed, ReadTimeout error"]])
-def test_scan_negative_read_timedout(execute_mock, expected_message, events, node):
+@pytest.mark.parametrize(("mode", "connection"), [["partition", i] for i in [AsyncOperationTimedOutConnection, AsyncConnectionExceptionConnection, AsyncReadTimeoutConnection]]+[[j, i] for j in ['aggregate', 'table'] for i in [OperationTimedOutConnection, ConnectionExceptionConnection, ReadTimeoutConnection]])
```

Comment:
why not use double parametrization here? This is very hard for me to udnerstand what parameters are actually tried